node/relay_request.js

```diff
@@ -51,11 +52,33 @@ RelayRequest.prototype.createOutRequest = function createOutRequest(host) {
         return;
     }
 
+    self.peer = self.channel.peers.choosePeer(null, {
+        host: host
+    });
+    if (!self.peer) {
+        self.onError(errors.NoPeerAvailable());
+        return;
+    }
```

Comment:
@Raynos not sure if it matters, since self.channel.request was passed in opts.host, which will lead to the path that uses self.peers.request.